tracer/tracer_test.go

```diff
@@ -158,24 +149,120 @@ func TestTracerEdgeSampler(t *testing.T) {
 	assert.Equal(count, tracer1.buffer.Len())
 }
 
-// Mock Transport with a real Encoder
-type DummyTransport struct {
-	pool *encoderPool
+func TestTracerConcurrent(t *testing.T) {
+	assert := assert.New(t)
+	tracer, transport := getTestTracer()
+
+	// Wait for three different goroutines that should create
+	// three different traces with one child each
+	var wg sync.WaitGroup
+	wg.Add(3)
+	go func() {
+		defer wg.Done()
+		tracer.NewRootSpan("pylons.request", "pylons", "/").Finish()
+	}()
+	go func() {
+		defer wg.Done()
+		tracer.NewRootSpan("pylons.request", "pylons", "/home").Finish()
+	}()
+	go func() {
+		defer wg.Done()
+		tracer.NewRootSpan("pylons.request", "pylons", "/trace").Finish()
+	}()
+
+	wg.Wait()
+	tracer.Flush()
+	traces := transport.Traces()
+	assert.Len(traces, 3)
+	assert.Len(traces[0], 1)
+	assert.Len(traces[1], 1)
+	assert.Len(traces[2], 1)
 }
 
-func (t *DummyTransport) Send(spans []*Span) (*http.Response, error) {
-	encoder := t.pool.Borrow()
-	defer t.pool.Return(encoder)
-	return nil, encoder.Encode(spans)
+func TestTracerConcurrentMultipleSpans(t *testing.T) {
```

Comment:
Tests the full flow with 2 goroutines using our tracer to create two traces with two spans each